ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/Footer.kt

```diff
@@ -0,0 +1,146 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.R
+import com.revenuecat.purchases.ui.revenuecatui.UIConstant
+import com.revenuecat.purchases.ui.revenuecatui.data.MockViewModel
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallViewModel
+import com.revenuecat.purchases.ui.revenuecatui.data.TestData
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.ColorsFactory
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
+import java.net.URL
+
+@Composable
+internal fun Footer(
+    templateConfiguration: TemplateConfiguration,
+    viewModel: PaywallViewModel
+) {
+    Footer(
+        configuration = templateConfiguration.configuration,
+        colors = templateConfiguration.getCurrentColors(),
+        viewModel = viewModel
+    )
+}
+
+@Composable
+internal fun Footer(
+    configuration: PaywallData.Configuration,
+    colors: TemplateConfiguration.Colors,
+    viewModel: PaywallViewModel
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(intrinsicSize = IntrinsicSize.Max)
+            .padding(
+                start = UIConstant.defaultHorizontalPadding,
+                end = UIConstant.defaultHorizontalPadding,
+                bottom = UIConstant.defaultButtonVerticalSpacing,
+            ),
+        horizontalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        val color = colors.text1
+
+        if (configuration.displayRestorePurchases) {
+            Button(
+                text = stringResource(id = R.string.restore_purchases),
+                color = color,
+                action = { viewModel.restorePurchases() }
+            )
+
+            if (configuration.termsOfServiceURL != null || configuration.privacyURL != null) {
+                Separator(color = color)
+            }
+        }
+
+        configuration.termsOfServiceURL?.let {
+            Button(
+                text = stringResource(id = R.string.terms_and_conditions),
+                color = color,
+                action = { viewModel.openURL(it) }
+            )
+
+            if (configuration.privacyURL != null) {
+                Separator(color = color)
+            }
+        }
+
+        configuration.privacyURL?.let {
+            Button(
+                text = stringResource(id = R.string.privacy_policy),
+                color = color,
+                action = { viewModel.openURL(it) }
+            )
+        }
+    }
+}
+
+@Composable
+private fun Separator(color: Color) {
+    Canvas(
+        modifier = Modifier
+            .size(10.dp),
```

Comment:
In iOS we use `@ScaledMetric` to have these "magic numbers" scale up with the font sizes. What's the appropriate way of doing this in Android to avoid this magic number?

Reply:
Oh interesting... I haven't actually done that before, and not sure if it's worth scaling the separators here... But out of curiosity and took a look and I think we need to calculate the `dp` value depending on the densitiy, kinda opposite of what they are triying to do in this SO post: https://stackoverflow.com/a/69112444

Reply:
I'll leave a TODO for that.